.github/workflows/main.yml

```diff
@@ -198,8 +188,55 @@ jobs:
           args: -Dproject.settings=.sonar-project.properties
 
 
-  # Ensure pull requests only have a single commit
+  # Lint checks that do not require compilation
+  lint:
+    name: Lint Checks
+    strategy:
+      fail-fast: false
+      matrix:
+        java_distribution: [ temurin ]
+        java_version: [ 17 ]
+        scala_version: [ 2.12.17 ]
+        os: [ ubuntu-22.04 ]
+    runs-on: ${{ matrix.os }}
+    env:
+      SBT: sbt -J-Xms1024m -J-Xmx5120m -J-XX:ReservedCodeCacheSize=512m -J-XX:MaxMetaspaceSize=1024m ++${{ matrix.scala_version }}
+    steps:
+
+      ############################################################
+      # Setup
+      ############################################################
+
+      - name: Check out Repository
+        uses: actions/checkout@v3.3.0
+        with:
+          fetch-depth: 0
```

Comment:
Can you add a comment why we're checking out with zero fetch-depth?

Reply:
It took some digging to figure out the reason why (our above comment just says to improve reporting, but I wasn't sure what that meant). I found out it's about sonar scanning, briefly documented here:

https://github.com/SonarSource/sonarcloud-github-action

And detailed reasons here:

https://community.sonarsource.com/t/git-fetch-depth-implications/75260

We don't run sonar scans in the lint checker (sonar scan uses coverage reports so needs to be in the main CI job), so I've removed the `fetch-depth: 0` here. I've also updated the above comment to mention sonar scan as the reason for it.